pages/docs/reference/compatibility.md

```diff
@@ -0,0 +1,114 @@
+---
+type: doc
+layout: reference
+category: "Compatibility"
+title: "Compatibility"
+---
+
+# Compatibility
+
+This page describes compatibility guarantees for different versions and subsystems of Kotlin. 
+
+## Compatibility glossary
+
+Compatibility is a function of a pair of versions, e.g. we can speak of compatibility between versions 1.2 and 1.1.5. In such a pair, one version is *older* than another when its smaller wrt version comparison, e.g. 1.1.5 is older than 1.2, even if it was released later. In the list below we explain available compatibility modes for such pairs of versions. We use OV for "Older Version", and NV for "Newer Version". 
+
+- **C** - Full **C**ompatibility
+  - Language
+    - no syntax changes (*modulo bugs**)
+    - new warnings/hints may be added or removed
+  - API (`kotlin-stdlib-*`, `kotlin-reflect-*`)
+    - no API changes
+    - deprecations with level `WARN` may be added/removed
+  - Binaries (ABI)
+    - runtime: binaries can be used interchangeably
+    - compilation: binaries can be used interchangeably
+- **BCL** - **B**ackward **C**ompatibility for the **L**anguage
+  - Language
+    - syntax deprecated in OV may be removed in NV
+    - other than that, all code compilable in OV is compilable by in NV (modulo bugs*)
+    - new syntax may be added in NV
+    - some restrictions of OV may be lifted in NV 
+    - new warnings/hints may be added or removed
+  - API (`kotlin-stdlib-*`, `kotlin-reflect-*`)
+    - new APIs may be added
+    - deprecations with level `WARN` may be added/removed
+    - deprecations with level `WARN` may be elevated to level `ERROR` or `HIDDEN` in NV
```

Comment:
1. level `WARNING`
2. declarations that have level `WARNING` in OV may be elevated to level `ERROR` or `HIDDEN` in NV